### DISCUSSION TITLE
<metadata name="net.sf.eclipsecs.core.lastEnabledSeverity" value="inherit"/>

### DISCUSSION THREAD

**Discussion by leerho** [checkstyle/checkstyle#10096] [Q&A]
What does the above metadata property do and why is it inserted into nearly all my modules in my checkstyle.xml?

It appears to be specific to Eclipse.  Yet, I would like to use my checkstyle configuration in other IDEs such as IntelliJ.

Can I remove it?

(I have extensively searched the Internet for answers and have not been successful in finding any :) ) 

Thanks!

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/blob/c25f0f271b770cd7fa08cbc894c68a7353321b56/src/main/java/com/puppycrawl/tools/checkstyle/ConfigurationLoader.java#L556
Metadata doesn't do anything in the checkstyle library, and has no impact on our library if removed.

It was probably added as a way for 3rd parties to do something custom, but I am not aware of anyone using it or how they would get to it without reusing and modifying our configuration loading classes.

**Reply by leerho** [reply]
Thanks, but this doesn't answer what it does. I appreciate that Checkstyle ignores "metadata" entries.  The answer clearly has to come from the Eclipse community.

**Reply by rnveach** [reply]
You can ask in eclipse-cs 's discussion area ( https://github.com/checkstyle/eclipse-cs/discussions ) . This discussion area is for the library only , and I doubt eclipse-cs maintainers will view here unless they are pinged.

Checkstyle library won't do anything with it. It appears eclipse-cs does have its own custom configuration reader ( https://github.com/checkstyle/eclipse-cs/blob/8f139aad405aa431b9ceb242b29abbfd438de3b1/net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/ConfigurationReader.java#L255 ) .
